tmail-backend/mailets/src/main/java/com/linagora/tmail/mailet/OpenAIMailet.java

```diff
@@ -0,0 +1,143 @@
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Strings;
+
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+
+/**
+ * OpenAIMailet is a mailet designed to automatically reply to emails
+ * sent to a specific recipient (e.g., gpt@linagora.com). It uses the OpenAI
+ * API to generate responses.
+ *
+ * <p>
+ * Configuration example:
+ *
+ * <pre>
+ * {@code
+ * <mailet match="com.linagora.tmail.mailet.RecipientsContain=gpt@linagora.com" class="com.linagora.tmail.mailet.OpenAIMailet">
+ *     <apiKey>demo</apiKey>
+ *     <gptAddress>gpt@tmail.com</gptAddress>
+ * </mailet>
+ * }
+ * </pre>
+ * </p>
+ *
+ * <p>
+ * The mailet configuration should include the following parameters:
+ * </p>
+ * <ul>
+ * <li><b>apiKey</b>: The API key for accessing the OpenAI service.</li>
+ * <li><b>gptAddress</b>: The email address used to send replies generated by the OpenAI model.</li>
+ * </ul>
+ */
+public class OpenAIMailet extends GenericMailet {
+    public static final String API_KEY_PARAMETER_NAME = "apiKey";
+    public static final String GPT_ADDRESS_PARAMETER_NAME = "gptAddress";
+
+    private ChatLanguageModel openAiModel;
+    private MailAddress gptAddress;
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        if (gptLoopDetected(mail)) {
+            return;
+        }
+
+        try {
+            String prompt = MimeMessageUtil.asString(mail.getMessage());
+            ChatMessage systemMessage = new SystemMessage("You are a helpful mailing chatbot. Respond to the following email as a chatbot, not as the recipient.");
+            ChatMessage userMessage = new UserMessage(prompt);
+            String answer = openAiModel.generate(systemMessage, userMessage)
+                .content()
+                .text();
+            MimeMessage replyMimeMessage = evaluateReplyMimeMessage(mail, answer);
+
+            getMailetContext()
+                .sendMail(gptAddress, evaluateRecipients(mail), replyMimeMessage);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean gptLoopDetected(Mail mail) {
+        return mail.getMaybeSender().equals(MaybeSender.of(gptAddress));
+    }
+
+    private MimeMessage evaluateReplyMimeMessage(Mail mail, String answer) throws MessagingException {
+        MimeMessage reply = (MimeMessage) mail.getMessage()
+            .reply(true);
+
+        stripGptFromRecipients(reply, Message.RecipientType.TO);
+        stripGptFromRecipients(reply, Message.RecipientType.CC);
+        stripGptFromRecipients(reply, Message.RecipientType.BCC);
+
+        reply.setSentDate(new Date());
+        reply.setFrom(gptAddress.asString());
+        reply.setText(answer);
```

Comment:
Ideally we should quote the previous answer if not too complex

```
Boil it for 10 minutes

On 15th july grandma <gm@toto.com> wrote:
> How to cook an egg?
```

If too complex I would agree with what had been done today.